### DISCUSSION TITLE
JS .focus() not working in PopUp? (even in the MinimalExample)

### DISCUSSION THREAD

**Discussion by SB97** [cefsharp/CefSharp#4457] [Q&A]
Hello,

i can`t get focus on a input field in a Popup. For comparison i tried the CefSharp.MinimalExample and have the same problem there.

The Call:
```HTML
<button class="button" onClick="window.open('test.php','fenster','popup');">
     <span class="icon">Open PopUp</span>
</button>
```

The Popup:
```HTML
<body>
<form action="/action_page.php">
  <label for="fname">First name:</label><br>
  <input type="text" id="fname" name="fname" value="John"><br>
  <label for="lname">Last name:</label><br>
  <input type="text" id="lname" name="lname" value="Doe"><br><br>
  <input type="submit" value="Submit">
</form> 

</body>

<script>
       document.getElementById("fname").focus();
</script>
```

My Code works in Firefox, Edge and Chrome.

Im at a loss. My code works but not in CefSharp, but i can`t imagine such a basic feature not working.

**Comment by amaitland** [comment]
More likely a `CEF` issue than it is a `CefSharp` specific one. Test with the `CEF Sample Application` as detailed in https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef

Making **sure** to use the command line args that match the implementation you are using.

Possibly related to https://github.com/chromiumembedded/cef/issues/2931

**Reply by aimcosoftware** [reply]
Tried this and it's a bit odd because `.select()` works but not `.focus()`

**Comment by SB97** [comment]
Thanks for the reply.

CEF Sample Application: works (does not flash the cursor in the input field)
CefSharp.MinimalExample (net472, net472, netcore): not working
Chrome (12. ...): works

All were started as is without any arguments.

**Reply by amaitland** [reply]
> All were started as is without any arguments.

You need to test with the arguments listed in the link for a meaningful result.

**Reply by SB97** [reply]
I'm sorry. I'm not sure what you mean exactly.

If it's the ` cefclient.exe --multi-threaded-message-loop --no-sandbox`
from https://github.com/cefsharp/CefSharp/blob/master/CONTRIBUTING.md#cefsharp-vs-chromium-embedded-frameworkcef :

the CEF Sample Application still works and now also shows the cursor.

**Reply by amaitland** [reply]
> - Extract and run cefclient.exe
      - If you are using WPF/OffScreen run
      ```
      cefclient.exe --multi-threaded-message-loop --no-sandbox --off-screen-rendering-enabled --enable-gpu
      ```
      - If you are using WinForms run
      ```
      cefclient.exe --multi-threaded-message-loop --no-sandbox
      ```

You need to test using the command line args specified based on the implementation you are using. You also haven't said if you are using `WinForms/WPF` which leaves me having to guess that you are using `WinForms`.

> All were started as is without any arguments.

Testing without arguments will compare to a totally different configuration which isn't the default.

Quick look at the `cefclient` source and you'll likely also need to test with the default popup behaviour which is what you get by default using `CefSharp`.

Use [--use-default-popup](https://github.com/chromiumembedded/cef/blob/4b3c3132cbaf6176d4bcc0cbe90db66fe99ad077/tests/shared/common/client_switches.cc#L53) in **addition** to the base set of command line args.

**Reply by SB97** [reply]
Thank you for the explanation.

Yes i use WinForms
`.\cefclient.exe  --multi-threaded-message-loop --no-sandbox --use-default-popup`
 the CEF Sample Application now behaves like the CefSharp Example, and .focus() is not working anymore.

What can I take away from this finding?

**Reply by amaitland** [reply]
How does it behave if you host the popup in your own `Form`?

https://github.com/cefsharp/CefSharp/wiki/General-Usage#winforms---hosting-popup-using-tab-control

**Reply by SB97** [reply]
I would like to test, but i don`t have a tabbed project.

But as there is proof of a problem already: Should i open a real issue?

**Reply by aimcosoftware** [reply]
@amaitland Tried this on my popup testing, which is in my own window, it doesn't seem to work.

**Reply by amaitland** [reply]
> I would like to test, but i don`t have a tabbed project.

You can use the same method to host the popup in a new Form created entirely in code.

> Should i open a real issue?

As the problem reproduce with the `CEF Sample Application` you will need to open an issue on https://github.com/chromiumembedded/cef/issues

Please make sure to search through open issues before creating a new one.

> Tried this on my popup testing, which is in my own window, it doesn't seem to work.

@aimcosoftware Does focusing another control outside the popup browser then refocusing the browser restore it's focus?

**Reply by aimcosoftware** [reply]
It's because the browser doesn't have focus, if you set focus to CEFSharp browser, it works.

I did it in `LoadingStateChanged`. e.g:

``` basic
Private Sub Browser_LoadingStateChanged(Sender As Object, e As LoadingStateChangedEventArgs) Handles Browser.LoadingStateChanged
	If Not e.IsLoading Then
		Invoke(Sub() Browser.Focus())
	End If
End Sub

```

**Reply by SB97** [reply]
> As the problem reproduce with the CEF Sample Application you will need to open an issue on https://github.com/chromiumembedded/cef/issues

https://github.com/chromiumembedded/cef/issues/3481
